sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/options/ContainerCreateOptions.java

```diff
@@ -0,0 +1,115 @@
+package com.azure.android.storage.blob.options;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.azure.android.core.annotation.Fluent;
+import com.azure.android.core.util.CancellationToken;
+import com.azure.android.storage.blob.models.PublicAccessType;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Extended options that may be passed when creating a container.
+ */
+@Fluent
+public class ContainerCreateOptions {
+
+    private final String containerName;
+    private Map<String, String> metadata;
+    private PublicAccessType publicAccessType;
+    private Integer timeout;
+    private CancellationToken cancellationToken;
+
+    /**
+     * @param containerName The container name.
+     */
+    public ContainerCreateOptions(@NonNull String containerName) {
+        Objects.requireNonNull(containerName);
+        this.containerName  = containerName;
+    }
+
```

Comment:
probably its wip, but just want to point Javadoc is in-complete for ctr, getters and setters

Reply:
Not sure if you commented this before or after I added some docs. Do you want me to add more detail than exists now? This docs are basically copy pasted from the Java SDK